include/conf/sockopts.h

```diff
@@ -20,6 +20,122 @@
+    DPVSMSG(DPVS_SO_GET_SERVICES)   \
+    DPVSMSG(DPVS_SO_GET_SERVICE)    \
+    DPVSMSG(DPVS_SO_GET_DESTS)      \
+    DPVSMSG(SOCKOPT_SVC_MAX)        \
+    \
+    DPVSMSG(SOCKOPT_SET_ROUTE_ADD)  \
+    DPVSMSG(SOCKOPT_SET_ROUTE_DEL)  \
+    DPVSMSG(SOCKOPT_SET_ROUTE_SET)  \
+    DPVSMSG(SOCKOPT_SET_ROUTE_FLUSH)\
+    DPVSMSG(SOCKOPT_GET_ROUTE_SHOW) \
+    \
+    DPVSMSG(SOCKOPT_SET_IFADDR_ADD)     \
+    DPVSMSG(SOCKOPT_SET_IFADDR_DEL)     \
+    DPVSMSG(SOCKOPT_SET_IFADDR_SET)     \
+    DPVSMSG(SOCKOPT_SET_IFADDR_FLUSH)   \
+    DPVSMSG(SOCKOPT_GET_IFADDR_SHOW)    \
+    \
+    DPVSMSG(SOCKOPT_NETIF_SET_LCORE)    \
+    DPVSMSG(SOCKOPT_NETIF_SET_PORT)     \
+    DPVSMSG(SOCKOPT_NETIF_SET_BOND)     \
+    DPVSMSG(SOCKOPT_NETIF_SET_MAX)      \
+    DPVSMSG(SOCKOPT_NETIF_GET_LCORE_MASK)   \
+    DPVSMSG(SOCKOPT_NETIF_GET_LCORE_BASIC)  \
+    DPVSMSG(SOCKOPT_NETIF_GET_LCORE_STATS)  \
+    DPVSMSG(SOCKOPT_NETIF_GET_PORT_LIST)        \
+    DPVSMSG(SOCKOPT_NETIF_GET_PORT_BASIC)       \
+    DPVSMSG(SOCKOPT_NETIF_GET_PORT_STATS)       \
+    DPVSMSG(SOCKOPT_NETIF_GET_PORT_EXT_INFO)    \
+    DPVSMSG(SOCKOPT_NETIF_GET_BOND_STATUS)      \
+    DPVSMSG(SOCKOPT_NETIF_GET_MAX)  \
+    \
+    DPVSMSG(SOCKOPT_SET_NEIGH_ADD)  \
+    DPVSMSG(SOCKOPT_SET_NEIGH_DEL)  \
+    DPVSMSG(SOCKOPT_GET_NEIGH_SHOW) \
+    \
+    DPVSMSG(SOCKOPT_SET_BLKLST_ADD) \
+    DPVSMSG(SOCKOPT_SET_BLKLST_DEL) \
+    DPVSMSG(SOCKOPT_SET_BLKLST_FLUSH)   \
+    DPVSMSG(SOCKOPT_GET_BLKLST_GETALL)  \
+    \
+    DPVSMSG(SOCKOPT_SET_WHTLST_ADD) \
+    DPVSMSG(SOCKOPT_SET_WHTLST_DEL) \
+    DPVSMSG(SOCKOPT_SET_WHTLST_FLUSH)   \
+    DPVSMSG(SOCKOPT_GET_WHTLST_GETALL)  \
+    \
+    DPVSMSG(SOCKOPT_SET_VLAN_ADD)   \
+    DPVSMSG(SOCKOPT_SET_VLAN_DEL)   \
+    DPVSMSG(SOCKOPT_GET_VLAN_SHOW)  \
+    \
+    DPVSMSG(SOCKOPT_TC_ADD)     \
+    DPVSMSG(SOCKOPT_TC_DEL)     \
+    DPVSMSG(SOCKOPT_TC_CHANGE)  \
+    DPVSMSG(SOCKOPT_TC_REPLACE) \
+    DPVSMSG(SOCKOPT_TC_SHOW)    \
+    \
+    DPVSMSG(SOCKOPT_SET_CONN)           \
+    DPVSMSG(SOCKOPT_GET_CONN_ALL)       \
+    DPVSMSG(SOCKOPT_GET_CONN_SPECIFIED) \
+    \
+    DPVSMSG(SOCKOPT_IP6_SET)    \
+    DPVSMSG(SOCKOPT_IP6_STATS)  \
+    \
+    DPVSMSG(SOCKOPT_TUNNEL_ADD) \
+    DPVSMSG(SOCKOPT_TUNNEL_DEL) \
+    DPVSMSG(SOCKOPT_TUNNEL_CHANGE)  \
+    DPVSMSG(SOCKOPT_TUNNEL_REPLACE) \
+    DPVSMSG(SOCKOPT_TUNNEL_SHOW)    \
+    \
+    DPVSMSG(SOCKOPT_SET_KNI_ADD)    \
+    DPVSMSG(SOCKOPT_SET_KNI_DEL)    \
+    DPVSMSG(SOCKOPT_SET_KNI_FLUSH)  \
+    DPVSMSG(SOCKOPT_GET_KNI_LIST)   \
+    \
+    DPVSMSG(SOCKOPT_SET_IPSET)      \
+    DPVSMSG(SOCKOPT_GET_IPSET_TEST) \
+    DPVSMSG(SOCKOPT_GET_IPSET_LIST) \
+    \
+    DPVSMSG(SOCKOPT_SET_ROUTE6_ADD_DEL) \
+    DPVSMSG(SOCKOPT_SET_ROUTE6_FLUSH)   \
+    DPVSMSG(SOCKOPT_GET_ROUTE6_SHOW)    \
+    \
+    DPVSMSG(SOCKOPT_SET_IFTRAF_ADD) \
+    DPVSMSG(SOCKOPT_SET_IFTRAF_DEL) \
+    DPVSMSG(SOCKOPT_GET_IFTRAF_SHOW)\
+
+typedef enum {
+    DPVSMSG_SOCKOPT_ENUM(ENUM_ITEM)
+}SOCKOPT_CODE;
```

Comment:
Would `SOCKOPT_CODE` type be used anywhere else? If not, I think an unnamed type looks more clearer.

```
enum {
    DPVSMSG_SOCKOPT_ENUM(ENUM_ITEM)
}
```